plugins/inputs/opcua_listener/subscribe_client.go

```diff
@@ -32,6 +32,40 @@ type SubscribeClient struct {
 	processingCancel context.CancelFunc
 }
 
+func AssignConfigValuesToRequest(req *ua.MonitoredItemCreateRequest, monParams *input.MonitoringParameters, log telegraf.Logger) {
+	req.RequestedParameters.SamplingInterval = float64(time.Duration(monParams.SamplingInterval) / time.Millisecond)
+
+	if monParams.QueueSize != nil {
+		req.RequestedParameters.QueueSize = *monParams.QueueSize
+	}
+
+	if monParams.DiscardOldest != nil {
+		req.RequestedParameters.DiscardOldest = *monParams.DiscardOldest
+	}
+
+	if monParams.DataChangeFilter != nil {
+		deadbandValue := monParams.DataChangeFilter.DeadbandValue
+		switch {
+		case deadbandValue == nil:
+			log.Warnf(
+				"No deadband value was set. No filter could be applied to '%s'.",
+				req.ItemToMonitor.NodeID)
+		case *deadbandValue < 0:
+			log.Warnf(
+				"Negative deadband value is not supported. No filter could be applied to '%s'.",
+				req.ItemToMonitor.NodeID)
```

Comment:
For both of these warnings, shouldn't we catch them during init as well and not start up?

Reply:
Ok, so I changed it in way that it fails if one of the `DataChangeFilter` parameters is not set correctly.